### DISCUSSION TITLE
better usability on desktop

### DISCUSSION THREAD

**Discussion by Croworbit** [lightspark/lightspark#1139] [General]
I prefer to run my flash games on the desktop, and the only option is to use CLI.
now, I have used CLI programs before and thats not the issue.
doing `lightspark swfname.swf` might work well to test a few files, but I have hundreds of swfs
some of them with really long or similar names
its really tedious and inconvenient to type out all the names of each SWF

if the desktop app had some sort of file picker, that would be much simpler and I would be able to test way more often
even if it just brought up my file manager, that would be way better, I could just click the file I want to try

I want to contribute by finding issues, but the current way is too tedious...

**Comment by Fancy2209** [comment]
it does! try the latest [pre-release](https://github.com/lightspark/lightspark/releases/tag/latest)
![imagem](https://github.com/user-attachments/assets/dd99cf78-3233-4d3c-b2e0-87822e222ea5)
you click the cog on the empty entry, specify a swf, and it gets added

**Reply by Croworbit** [reply]
that's not really what I expected, but I'll try it
so this acts like a 'bookmark' system then, huh?

I guess I didn't notice because I avoid betas, pre releases, and early accesses

**Reply by Fancy2209** [reply]
> that's not really what I expected, but I'll try it so this acts like a 'bookmark' system then, huh?
> 
> I guess I didn't notice because I avoid betas, pre releases, and early accesses

Next Stable Release will happen god knows when, pre-release is actually just the latest change and should be fine to use
Yeah right now it's just bookmarks, iirc there are plans for a better UI down the line

**Reply by Fancy2209** [reply]
Also you can just drag a swf into the exe if you don't need to change options

**Reply by Croworbit** [reply]
> Also you can just drag a swf into the exe if you don't need to change options

I don't use windows for flash things

**Reply by Fancy2209** [reply]
> > Also you can just drag a swf into the exe if you don't need to change options
> 
> I don't use windows for flash things

then how are you using the stable release of lightspark? For linux you have to build it from source anyways

**Reply by Croworbit** [reply]
yes, I compiled lightspark many months ago
is it really that surprising?

**Reply by Fancy2209** [reply]
> yes, I compiled lightspark many months ago is it really that surprising?

No I just got confused, my bad.
If you compile on the latest commit you should get the launcher if you open it without args, sorry if I am beeing confusing.
